packages/studio-plugin/src/createStudioStylingPlugin.ts

```diff
@@ -0,0 +1,47 @@
+import upath from "upath";
+import { PluginOption } from "vite";
+/**
+ * Adds the ?inline query parameter to all CSS imports
+ * unless they are CSS from Studio, which is determined
+ * from the path of its importer. This prevents user styling
+ * from affecting Studio UI.
+ */
+export default function createStudioStylingPlugin(): PluginOption {
+  return {
+    name: "yext-studio-style-plugin",
+    enforce: "pre",
+    resolveId(id, importer) {
+      if (!importer) {
+        return;
+      }
+      if (isImportedByStudio(importer)) {
+        return;
+      }
+      if (isStyleFile(id)) {
+        if (upath.isAbsolute(id)) {
+          return addQueryParameter(id, "inline");
+        }
+        return upath.join(
+          upath.dirname(importer),
+          addQueryParameter(id, "inline")
+        );
+      }
+    },
+  };
+}
+
+function isImportedByStudio(id: string) {
+  return !!(id.match(/.*@yext\/studio.*/) || id.match(/.*packages\/studio*/));
```

Comment:
I'm not a huge fan of the way this is currently done-- it's just a regex checking the path of what imported the file.  I'm not sure what we can assume.